hil-test/tests/spi_full_duplex_dma.rs

```diff
@@ -121,6 +121,10 @@ mod tests {
         let transfer = spi.dma_transfer(&mut send, &mut receive).unwrap();
         transfer.wait().unwrap();
         assert_eq!(send[0], receive[0]);
+        assert_eq!(send[1], receive[1]);
+        // Read the 2 remaining bytes so it does not cause issues in the next test.
```

Comment:
what issues does it cause if we don't repeat the transfer?

Reply:
that's very weird 🤷‍♂️

Reply:
This test was passing before, it should not need to be changed. This is either a bug for the S2 implementation or an issue with the new runners, I'd imagine. Not really excited about merging hacks into our HIL tests...

Reply:
This should not be related to the new runner, the old runner also required this change and the same behavior happens locally. I've reverted the changes and if we merge this, we should open an issue to investigate `spi_fullduplex_dma` test on S2 which fails in 2 different ways, and its currently disabled.

Reply:
I would prefer all HIL tests to be passing for all devices before merging this personally, but sure whatever works.